### PR TITLE
[WIP] Check point PR for migration of Model Tests

### PR DESCRIPTION
**Pull request by mtrue01**
This is a check point PR for the work I have currently done for the model migration

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/AbstractKeycloakTest.java [hunk 1]
```diff
@@ -231,12 +231,13 @@ public void afterAbstractKeycloakTest() {
                 }
             }
             testContext.getCleanups().clear();
-        }
 
-        postAfterAbstractKeycloak();
 
-        // Remove all browsers from queue
-        DroneUtils.resetQueue();
+            postAfterAbstractKeycloak();
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/AbstractKeycloakTest.java]: I am not sure about this change and why is it needed? I think the "postAfterAbstractKeycloak()" and "DroneUtils.resetQueue()" need to be called also when "importAfterEachMethod" is true? Which is changed here.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/rule/KeycloakRule.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
+/*
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/rule/KeycloakRule.java]: I think this file shouldn't be added to the new testsuite.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/rule/ErrorServlet.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
+/*
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/rule/ErrorServlet.java]: I think this file shouldn't be added to the new testsuite. And it seems it is not needed for any model test?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/rule/AbstractKeycloakRule.java [hunk 4]
```diff
@@ -0,0 +1,399 @@
+/*
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/rule/AbstractKeycloakRule.java]: I think this file shouldn't be added to the new testsuite. And it seems it is not needed for any model test?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,762 @@
...
+        RealmModel realm = session.realms().getRealmByName("test");
+        UserSessionModel[] sessions = createSessions(session);
+
+        assertSession(session.sessions().getUserSession(realm, sessions[0].getId()), session.users().getUserByUsername("user1", realm), "127.0.0.1", started, started, "test-app", "third-party");
+        assertSession(session.sessions().getUserSession(realm, sessions[1].getId()), session.users().getUserByUsername("user1", realm), "127.0.0.2", started, started, "test-app");
+        assertSession(session.sessions().getUserSession(realm, sessions[2].getId()), session.users().getUserByUsername("user2", realm), "127.0.0.3", started, started, "test-app");
+    }
+
+    @Test
+    public void testUpdateSession() {
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/UserSessionProviderTest.java]: As we discussed, I am not a fan of "doubled" methods (each test method having separate static method). Is it possible to use @ModelTest instead and pass KeycloakSession as an argument to the method? Same for other places?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/ImportTest.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    @Test
+    public void demoDelete() throws Exception {
+        // was having trouble deleting this realm from admin console
+            removeRealm("demo-delete");
+    }
+    
+	@Test
+    public void install2() throws Exception {
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/ImportTest.java]: Is it possible to use @ModelTest instead and pass KeycloakSession as an argument to the method? Same for other places?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/CacheTest.java [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+	            realm = session.realms().getRealmByName("test");
+	            Assert.assertEquals(200, realm.getAccessCodeLifespanLogin());
+	            testApp = session.realms().getClientById(appId, realm);
+	            Assert.assertFalse(testApp.isEnabled());
+	        
+	        }
+		 });
+		 }
+    @Test
+    public void testAddUserNotAddedToCache() {
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/CacheTest.java]: Is it possible to use @ModelTest instead and pass KeycloakSession as an argument to the method? Same for other places?

**Review by mposolda — COMMENTED** [review]
@mtrue01 Thanks for the checkpoint PR. I've added few items to the feedback.

**Comment by mtrue01** [comment]
Yeah! I didn’t realize the rules files were in the PR.  I used @ModelTest after you had given it to me but the ones that do not were done using either the testingClient lambda or the double method thing.  The before, after, and helper functions all need to be called similarly and using @ModelTest with those did not seem to work the way I had anticipated because the behavior is different when called with @ModelTest and the double method I am using now.

Thoughts?

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/ImportTest.java (hunk 6 above)
Comment (mtrue01) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/ImportTest.java]: This file breaks way too easily, I would rather not mess with it.  I have removed the double static calls from the other files but I would like to leave this one alone.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/CacheTest.java (hunk 7 above)
Comment (mtrue01) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/CacheTest.java]: I spent too much time writing this file using testingClient and trying to figure out the appropriate patterns for using it.  I'd rather leave it as is, there are no double static calls.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/rule/LoggingRule.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+import org.jboss.logging.Logger;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class LoggingRule implements TestRule {
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/rule/LoggingRule.java]: I think this file don't need to be added to the new testsuite as well. Sorry, I probably did not point this file in previous review, but it is same/similar thing like all the other files (AbstractKeycloakRule, KeycloakRule etc). In the new testsuite, there is very similar thing (logging before/after test method) provided by the class TestEventsLogger, so no need to have same thing again in the new testsuite.

**Review by mposolda — COMMENTED** [review]
Looks quite good, thanks a lot for the updates!

Had just few more comments:
- LoggingRule shouldn't be needed in the new testsuite (see inline comment)
- Jenkins CI failed. I am leaving to you to figure the reason. It is possible that the failure is some jenkins stability issue and it's completely unrelated to this PR, but not sure...

**Review by mposolda — APPROVED** [review]
@mtrue01 Thanks!

**Pull request merged**